### PR TITLE
Require a space before first argument of a method call

### PR DESCRIPTION
**Pull request by cjlarose**
The Ruby parser appears to be completely fine with something like

```ruby
puts'hello'
```

The cop `Style/SpaceBeforeFirstArg` claims to check that "exactly one space is used between a method name and the first argument for method calls without parentheses", but does not account for source code that has zero spaces between the method name and the first argument.

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Used the same coding conventions as the rest of the project.
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [ ] All tests are passing.
* [X] The new code doesn't generate RuboCop offenses.
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [X] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by rchasman** [comment]
:dolphin:

**Comment by rchasman** [comment]
^ LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## master (unreleased)
 
+### New features
```
Comment (Drenmi) [inline CHANGELOG.md]: There's a section called `Changes` where we normally put changes to existing cops, although this might even qualify as a bug fix. 🙂

**Comment by Drenmi** [comment]
Great first contribution! ❤️

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### Bug fixes
 
+* [#4265](https://github.com/bbatsov/rubocop/pull/4265): Require a space before first argument of a method call in `Style/SpaceBeforeFirstArg` cop. ([@cjlarose][])
```
Comment (bbatsov) [inline CHANGELOG.md]: You have to add yourself to the bottom of the changelog.

Comment (cjlarose) [inline CHANGELOG.md]: Sorry about that! Updated the CHANGELOG with a link to myself.

**Comment by bbatsov** [comment]
👍

**Pull request merged**